pkg/domain/infra/abi/images.go

```diff
@@ -858,14 +858,15 @@ func execSu(execUser *user.User, command []string) error {
 	return cmd.Run()
 }
 
-func execMachine(execUser *user.User, command []string, machinectl string) error {
-	verb := machinectl
-	args := []string{"shell", "-q", execUser.Username + "@.host"}
-	if execUser.Uid == "0" {
-		args = append([]string{verb}, args...)
-		verb = "sudo"
+func execRunUser(execUser *user.User, command []string, runUser string) error {
+	cmdLogin, err := utils.LoginUser(execUser.Username)
+	if err != nil {
+		return err
 	}
-	cmd := utils.CreateSCPCommand(exec.Command(verb, args...), command)
-	logrus.Debugf("Executing via machinectl: %q", cmd)
+	defer cmdLogin.Process.Kill()
+	verb := runUser
+	args := []string{"-l", execUser.Username, "-c"}
+	cmd := utils.CreateSCPCommand(exec.Command(verb, args...), []string{strings.Join(command, " ")})
```

Comment:
completely untested, but wouldn't something like the following:

```go
	cmd := utils.CreateSCPCommand(args, []string{strings.Join(command, " ")})
	cmd.SysProcAttr = &syscall.SysProcAttr{
		Credential: &syscall.Credential{
			Uid:         execUser.Uid,
			Gid:         execUser.Gid,
			Groups:      nil,
			NoSetGroups: false,
		},
	}
```

run the command with the specified UID/GID without requiring `runuser`?

Reply:
Wasn't that leak already in `CreateSCPCommand`?

AFAICS `runuser` doesn't clean up the environment:

```
$ sudo XDG_RUNTIME_DIR=wrong runuser -u gscrivano printenv XDG_RUNTIME_DIR
wrong
```

In any case, you are right, and I think we need to drop the Environ block.

Reply:
I agree we should block the environment.

Reply:
What do you mean the environment is leaked? Doesnt -l clear lol env variables except for those specified? @Luap99 `clears all the environment variables except for TERM and variables specified by --whitelist-environment`

Reply:
I think -u and -l function differently according to the man pages

Reply:
Yes `-l` will create a clean env, I was commenting on the suggestion from @giuseppe

Reply:
ah okay, sorry I was looking at this on mobile. From what i have tried over the iterations, directly calling a n exec.Cmd via a different user using the sysprocattr basically never works and always leaves something behind from the previous session.

Reply:
can you specify some Env not nil?  From the doc:

```go
	// If Env is nil, the new process uses the current process's
	// environment.
```

Reply:
hmm maybe I should try to keep $TERM like runuser does and see if that works?

Reply:
I would set $TERM and $PATH

Reply:
seems to work, I am getting a storage error regarding the graph driver: `ERRO[0000] User-selected graph driver "vfs" overwritten by graph driver "overlay" from database - delete libpod local files to resolve` but the command finishes successfully. This could be my machine so I will push here and let the tests run